core/src/main/java/hudson/model/AutoCompletionCandidates.java

```diff
@@ -173,4 +173,23 @@ private String contextualNameOf(Item i) {
     private static boolean startsWithImpl(String str, String prefix, boolean ignoreCase) {
         return ignoreCase ? StringUtils.startsWithIgnoreCase(str, prefix) : str.startsWith(prefix);
     }
+
+    /**
+     * Auto-completes possible job names with support for folder suggestions.
+     * Inspired from hudson.plugins.parameterizedtrigger.BuildTriggerConfig
+     * @param value The value the user has typed in.
+     * @param container The nearby contextual {@link ItemGroup} to resolve relative job names from.
+     */
+
+    public static AutoCompletionCandidates ofJobNames(final String value, ItemGroup container) {
```

Comment:
Why are we not using the old function?

Reply:
There are 3 ways autocomplete was implemented
The oldest and worst - RunParameterDefinition - It has no folder support or context
Better - BuildTrigger - It has no folder support but it has context
The newest in Parametrized Build Trigger - It has folder support and context
In the previous PR segalaj, the commiter wanted to implement the 2nd option. Oleg said that it is not enough, it must support folder. I wrote the new function because the old one does not support folder suggestions and this one does. I got inspired from the 3rd option.